decode.go

```diff
@@ -1000,3 +1005,135 @@ func (d *Decoder) Decode(v interface{}) error {
 	}
 	return nil
 }
+
+type unmarshalableStringValue string
```

Comment:
You should add test code to `decode_test.go` **NOT** `decode.go`

Reply:
oh